lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1672,6 +1677,50 @@ void FIRStmtParser::emitInvalidate(Value val, Flow flow) {
       });
 }
 
+void FIRStmtParser::emitConnect(ImplicitLocOpBuilder &builder, Value dst,
+                                Value src) {
+  auto type = dst.getType().cast<FIRRTLType>();
+  if (!type.containsAnalog()) {
+    builder.create<ConnectOp>(dst, src);
+  } else if (type.isa<AnalogType>()) {
+    builder.create<AttachOp>(SmallVector{dst, src});
+  } else if (auto bundle = type.dyn_cast<BundleType>()) {
```

Comment:
Consider using `TypeSwitch` here as this is a repeated `isa`/`dyn_cast` pattern. Not a blocker.

Reply:
I didn't use TypeSwitch here since the first check `if (!type.containsAnalog())` is not compatible. I could have nested if-statements (or used an early break) but I thought this was easier to understand.

Reply:
Yeah, I was thinking to keep it as `if (isAnalog) { /* ... stuff .. */ ; continue }` and then drop into a type switch.  It's not an issue and do whatever you think makes sense here.